src/Sarif.Viewer.VisualStudio/ProjectNameLookup.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Collections.Generic;
+
+namespace Microsoft.Sarif.Viewer
+{
+    public sealed class ProjectNameLookup
+    {
+        private readonly static ProjectNameLookup _instance = new ProjectNameLookup();
+        private Dictionary<string, string> _projectNames = new Dictionary<string, string>();
+
+        static ProjectNameLookup() { }
+        private ProjectNameLookup() { }
+
+        public static ProjectNameLookup Instance => _instance;
+
+        public void SetName(string fileName)
+        {
+            if (_projectNames.ContainsKey(fileName))
+                return;
+
+            var project = SarifViewerPackage.Dte.Solution.FindProjectItem(fileName);
+            if (project?.Properties != null && project?.ContainingProject != null)
```

Comment:
>Properties [](start = 25, length = 10)

Why is it necessary to test `Properties`? You don't seem to use it in the subsequent code.

Reply:
This check (project.Properties != null) existed in the previous code.  I wasn't sure of repercussions of removing, so I left it in.